### DISCUSSION TITLE
Sending a new transaction to MetaMask for approval from within a MetaMask Snap

### DISCUSSION THREAD

**Discussion by nathanmm88** [MetaMask/snaps#2049] [Q&A] [answered]
Hi currently working on a little project and experimenting with MetaMask Snaps.

I want to be able to send a transaction to MetaMask from within a snap so that the user can approve just like we would if it originated from a dApp. 

Firstly, is this possible from within a MetaMask snap? and secondly how would I go about doing this? If anyone could share any examples where this has been done that would be great. I have worked through some of the other examples in the github repo to work with other bits of functionality but haven't seen anything on this.

Thanks in advance!

**Comment by Montoya** [comment, marked as answer]
It is not possible yet. When it becomes possible, we will share example code for it.

**Reply by nathanmm88** [reply]
That would be great, thank you. Do you have a rough idea on when this maybe implemented?

**Reply by 1337-ishaan** [reply]
Is this possible anytime Q2 2024?
